src/pyhf/parameters/paramview.py

```diff
@@ -1,114 +1,107 @@
-            else:
-                self._indices_concatenated = cat
-
-        else:
-            self._indices_concatenated = None
-
-        last = 0
-        sl = []
-        for s in [
-            self._par_map[x]['slice'].stop - self._par_map[x]['slice'].start
-            for x in (name if isinstance(name, list) else [name])
-        ]:
-            sl.append(slice(last, last + s))
-            last += s
-        self._slices = sl
+        batch = shape[0] if len(shape) > 1 else None
+
+        fullsize = default_backend.product(default_backend.astensor(shape))
+        flat_indices = default_backend.astensor(range(int(fullsize)), dtype='int')
+        self._all_indices = default_backend.reshape(flat_indices, shape)
+
+        # a tensor viewer that can split and stitch parameters
+        self.allpar_viewer = _tensorviewer_from_parmap(par_map, batch)
+
+        # a tensor viewer that can split and stitch the selected parameters
+        self.selected_viewer = _tensorviewer_from_slices(
+            [par_map[s]['slice'] for s in selection], selection, batch
+        )
 
         self._precompute()
         events.subscribe('tensorlib_changed')(self._precompute)
 
     def _precompute(self):
         tensorlib, _ = get_backend()
-        if self._indices_concatenated is not None:
-            self.indices_concatenated = tensorlib.astensor(
-                self._indices_concatenated, dtype='int'
-            )
 
-    def __repr__(self):
-        return '({} with [{}] batched: {})'.format(
-            self.tensor_shape,
-            ' '.join(list(self._par_map.keys())),
-            bool(self.batch_shape),
+        self.all_indices = tensorlib.astensor(self._all_indices)
+        (
+            self.index_selection,
+            self.stitched,
+            self.indices_concatenated,
+        ) = extract_index_access(
+            self.allpar_viewer, self.selected_viewer, self.all_indices
         )
 
-    @property
-    def index_selection(self):
-        """
-        Returns:
-            indices into parameter field accordig to requested subset of parameters
-            list of (batch_size, parset_size) tensors
-        """
-        return self._index_selection
-
-    @property
-    def slices(self):
-        """
-        Returns:
-            list index slices to retrieve a subset of the requested parameters
-        """
-        return self._slices
-
-    def get(self, tensor, indices=None):
-        """
-        Args:
-            tensor (`tensor`): the data tensor to extract a view from
-            indices (`tensor`): an optional index selection (default behavior is to use self.indices_concatenated)
-        Returns:
-            filtered set of parameter field/array :
-                type when batched: (sum of slice sizes, batchsize) tensor
-                type when not batched: (sum of slice sizes, ) tensors
-        """
+    def get(self, data, indices=None):
```

Comment:
which indices are being passed in here? the all_indices which can be > npars (if batched) or corresponding to the pars, and tiled across all batched?